google/gcloud/gcloud.go

```diff
@@ -42,90 +54,45 @@ func CommandWithConfigKey(v string) Option {
 
 func New(l log.Logger, cache cache.Cache, opts ...Option) (*GCloud, error) {
 	inst := &GCloud{
-		l:         l,
-		cache:     cache.Get("gcloud"),
-		configKey: "gcloud",
+		l:                    l,
+		cache:                cache.Get("gcloud"),
+		configKey:            "gcloud",
+		accountFileNameRegex: regexp.MustCompile(`(\w+)@(\w+)-(\w+)\.json`),
 	}
 	for _, opt := range opts {
 		if opt != nil {
-			if err := opt(inst); err != nil {
-				return nil, err
-			}
+			opt(inst)
 		}
 	}
-	if err := viper.UnmarshalKey(inst.configKey, &inst.cfg); err != nil {
-		return nil, err
-	}
-
-	if err := os.MkdirAll(inst.cfg.ConfigDir, 0o700); err != nil {
-		return nil, errors.Wrapf(err, "failed to create directory %q", inst.cfg.ConfigDir)
-	}
-
-	return inst, nil
-}
-
-func (gc *GCloud) ParseAccounts(ctx context.Context) ([]GCloudAccount, error) {
-	accountFiles, err := files.Find(ctx, gc.cfg.ConfigDir, "*.json")
-	if err != nil {
-		return nil, err
-	}
 
-	var accounts []GCloudAccount
-	for _, f := range accountFiles {
-		matchString := gcloudAccountFileNameRegex.FindAllStringSubmatch(filepath.Base(f), 1)
-		if len(matchString) == 0 {
-			continue
-		}
-		match := matchString[0]
-		acc := GCloudAccount{
-			Role:        match[1],
-			Environment: match[2],
-			Cluster:     match[3],
-			Path:        f,
+	if inst.cfg == nil && inst.configKey != "" {
+		if err := viper.UnmarshalKey(inst.configKey, &inst.cfg); err != nil {
+			return nil, err
 		}
-		accounts = append(accounts, acc)
-	}
-
-	return accounts, err
-}
-
-func (gc *GCloud) FindAccounts(ctx context.Context, env, cluster string) ([]GCloudAccount, error) {
-	accounts, err := gc.ParseAccounts(ctx)
-	if err != nil {
-		return nil, err
 	}
 
-	filtered := accounts[:0]
-	for _, acc := range accounts {
-		if acc.Environment == env && acc.Cluster == cluster {
-			filtered = append(filtered, acc)
+	// set config path to encapsuplte any mishaps global gcloud usage!
+	if inst.cfg.Login {
+		// ensure config path
+		if err := files.MkdirAll(inst.cfg.ConfigPath); err != nil {
```

Comment:
Still thing os.MkdirAll is a better fit. Why wrap it?

Reply:
also I want to ensure there, that we are within some (PROJECT_ROOT) so we're not going outside